File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        freq = {}
        for i in nums:
            if i not in freq:
                freq[i] = 1
            else:
                freq[i] += 1
            if freq[i] >= 2:
                return True
        return False

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let numSet = new Set(nums)
        let maxSeqCount = 0
        for(let num of numSet){
            if(!numSet.has(num - 1)){
                let currentNum = num
                let currentStreak = 1
                while(numSet.has(currentNum+1)){
                    currentNum++
                    currentStreak++
                }
                if(currentStreak>maxSeqCount){
                    maxSeqCount = currentStreak
                }
            }
            
        }
        return maxSeqCount
    }
}
